crates/bin/prove_block/src/state_utils.rs

```diff
@@ -14,9 +16,21 @@ use starknet_os::starknet::business_logic::fact_state::state::SharedState;
 use starknet_os::starkware_utils::commitment_tree::base_types::TreeIndex;
 // trait for Patricia Tree
 use starknet_os::starkware_utils::commitment_tree::binary_fact_tree::BinaryFactTree;
+use starknet_os::starkware_utils::commitment_tree::patricia_tree::patricia_tree::PatriciaTree;
 use starknet_os::storage::storage::FactFetchingContext;
 
-use crate::rpc_utils::CachedRpcStorage;
+use crate::rpc_utils::{CachedRpcStorage, RpcStorage};
+use crate::utils::get_subcalled_contracts_from_tx_traces;
+
+#[derive(Clone)]
+pub struct ProcessedStateUpdate {
+    // TODO: Use more descriptive types
+    pub address_to_class_hash: HashMap<Felt252, Felt252>,
+    pub address_to_nonce: HashMap<Felt252, Felt252>,
+    pub class_hash_to_compiled_class_hash: HashMap<Felt252, Felt252>,
+    pub storage_updates: HashMap<Felt252, HashMap<Felt252, Felt252>>,
+    pub accessed_addresses: HashSet<Felt252>,
+}
```

Comment:
Unrelated to your PR, but yeah we need to use better typing here. This should be doable once I get my dependency PR to work.